superset-frontend/src/chart/ChartRenderer.jsx

```diff
@@ -222,13 +222,22 @@ class ChartRenderer extends React.Component {
       queryResponse,
     } = this.props;
 
+    let chartClassName = snakeCase(vizType);
+    // It's bad practice to use unprefixed `vizType` as classnames for chart
+    // container. It may cause css conflicts as in the case of table chart.
+    // When migrating legacy chart types, we should gradually add the prefix
+    // `superset-chart-` to each one of them.
+    if (vizType === 'table') {
```

Comment:
could use `const` and ternary 

```ts
const chartClassName = vizType === 'table' ? `superset-chart-${chartClassName}` : snakeCase(vizType);
```

Reply:
Was planning to allow more viz types in the future.

```
 if (vizType === 'table' || vizType === 'another-one' ) {
```

Reply:
you could still use ternary for that when adding more `vizType` or use set if the list grows longer.
```ts
const visSet = new Set([...]);
const chartClassName = visSet.has(vizType) ? ... : ... ;
```

Reply:
Makes sense, will update.

Reply:
Oh, I think another reason why I didn't use ternary is that the new class is based on an existing variable. It just feels more natural to use if statements to override it than either creating two variables or call `snakeCase(vizType)` twice.

```js
let chartClassName = snakeCase(vizType);
if (vizType === 'table') {
  chartClassName = `superset-chart-${chartClassName}`;
}
```

vs

```js
let chartClassName = snakeCase(vizType);
chartClassName = vizType === 'table' ? `superset-chart-${chartClassName}` : chartClassName;
```

Still thinks the if block is more readable... 😬 

vs 

```js
const chartClassName = vizType === 'table' ? `superset-chart-${snakeCase(vizType)}` : snakeCase(vizType);
```

Reply:
CI has passed. Can we merge as is?